src/WebForms/Configuration/SiteMapSection.cs

```diff
@@ -32,94 +34,56 @@ namespace System.Web.Configuration;
-
-    private static readonly ConfigurationProperty _propDefaultProvider =
-        new ConfigurationProperty("defaultProvider",
-                                    typeof(string),
-                                    "AspNetXmlSiteMapProvider",
-                                    null,
-                                    null,
-                                    ConfigurationPropertyOptions.None);
-
-    private static readonly ConfigurationProperty _propEnabled =
-        new ConfigurationProperty("enabled",
-                                    typeof(bool),
-                                    true,
-                                    ConfigurationPropertyOptions.None);
-
-    private static readonly ConfigurationProperty _propProviders =
-        new ConfigurationProperty("providers",
-                                    typeof(ProviderSettingsCollection),
-                                    null,
-                                    ConfigurationPropertyOptions.None);
-
     private SiteMapProviderCollection _siteMapProviders;
+    private readonly SiteMapOptions _siteMapOptions;
 
 #pragma warning disable CA1810 // Initialize reference type static fields inline
     static SiteMapSection()
 #pragma warning restore CA1810 // Initialize reference type static fields inline
     {
         // Property initialization
-        _properties = [_propDefaultProvider, _propEnabled, _propProviders];
+       // _properties = [_propDefaultProvider, _propEnabled, _propProviders];
     }
 
-    public SiteMapSection()
+    public SiteMapSection(IOptions<SiteMapOptions> options)
     {
+        _siteMapOptions = options.Value;
     }
 
-    protected override ConfigurationPropertyCollection Properties
-    {
-        get
-        {
-            return _properties;
-        }
-    }
-
-    [ConfigurationProperty("defaultProvider", DefaultValue = "AspNetXmlSiteMapProvider")]
-    [StringValidator(MinLength = 1)]
     public string DefaultProvider
     {
         get
         {
-            return (string)base[_propDefaultProvider];
-        }
-        set
-        {
-            base[_propDefaultProvider] = value;
+            return _siteMapOptions.DefaultProvider;
         }
+
     }
 
-    [ConfigurationProperty("enabled", DefaultValue = true)]
     public bool Enabled
     {
         get
         {
-            return (bool)base[_propEnabled];
-        }
-        set
-        {
-            base[_propEnabled] = value;
+            Debug.Assert(_siteMapOptions.Enabled != null, "_siteMapOptions.Enabled != null");
+            return _siteMapOptions.Enabled.Value;
         }
+
     }
 
-    [ConfigurationProperty("providers")]
     public ProviderSettingsCollection Providers
     {
         get
         {
-            if (base[_propProviders] == null || ((ProviderSettingsCollection)base[_propProviders]).Count == 0)
+            if (_siteMapOptions.Providers == null || _siteMapOptions.Providers.Count == 0)
```

Comment:
can you ensure there is a default value in `_siteMapOptions.Providers` that is `[]`?

Reply:
need more info.. could you write a snippet what do you mean ?